src/pages/home/report/ReportActionCompose.js

```diff
@@ -29,6 +27,9 @@ const propTypes = {
     // The ID of the report actions will be created for
     reportID: PropTypes.number.isRequired,
 
+    // Should the input focus on mount
+    autoFocus: PropTypes.bool.isRequired,
```

Comment:
Just jumping in here for a quick second when I saw this. I think this should be an internal detail of `<TextInputFocusable>`. Is it possible to move it all into there and have it be contained?

Reply:
It already kind of is - `TextInputFocusable` passes props to `TextInput` which is configurable by `autoFocus` 
Do you want to update the prop types definition of `<TextInputFocusable>` to contain `autoFocus` or something else?

This is a flag that is only used when the component mounts and is handled by `TextInput` internally
The default is `false` 

Do you want to add some logic inside `TextInputFocusable` to set a different default value?

Whether a field should `autoFocus` (on mount) should be set from outside as it depends on context/usage. Trying to confine the logic inside `TextInputFocusable` would make it depend on (more and more) external information to make that decision - am I big screen, is something else focused, etc.

Even if `TextInputFocusable` have some opinion on the `autoFocus` prop it should still be overridable from outside

Example: A screen renders 2 (or 10) `TextInputFocusable` components which one would take the focus - it would be straightforward when you can set it from outside

Reply:
OK, I think I misunderstood the usage of this prop slightly, thank you for adding that context. 

My main concern was seeing `autoFocus={!props.isSmallScreenWidth}` and I assumed the prop was only being added to disable autoFocus on small screens. If that was the only application of the prop, then that's why I suggested to move that logic to be contained internally in the component itself.

For the case of multiple components being rendered, I can now understand how it's useful to have it be a prop. 

I have a few suggestions to clear this up a little:

1. Remove the logic you have for the screen width. I see that you added a comment saying it was just an example, and we shouldn't commit code that's just an example.
2. Rename this prop to something which more accurately reflects its usage. Perhaps `shouldFocusOnMount`
3. Add more context to the inline comment for this prop to explain the situation about rendering multiple components and needing to control which one has focus.

Reply:
Allright, I'll add it as a propType on the `TextInputFocusable` as well as it's passed there too due to the prop spread
 
About the name I left it `autoFocus` as it mirrors the react native's own text input behavior and name
- https://reactnative.dev/docs/textinput#autofocus
Also in `React` and html the input attribute name is `autoFocus` (autofocus for html) and it has the same behavior as well
So just for that sake as this behavior (focus on mount) is widely recognized as autoFocus please confirm you want me to change the name

Reply:
OK, `autoFocus` will be all right and the comment does a good enough job explaining what it's for. Thanks!